itest/rust/src/object_tests/validate_property_test.rs

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use godot::builtin::{Array, Dictionary, GString, StringName};
+use godot::classes::IObject;
+use godot::global::PropertyUsageFlags;
+use godot::meta::PropertyInfo;
+use godot::obj::NewAlloc;
+use godot::register::{godot_api, GodotClass};
+use godot::test::itest;
+
+#[derive(GodotClass)]
+#[class(base = Object, init)]
+pub struct ValidatePropertyTest {
+    #[export]
+    my_var: i64,
+}
+
+#[godot_api]
+impl IObject for ValidatePropertyTest {
+    fn validate_property(&self, property: &mut PropertyInfo) {
+        if property.property_name.to_string() == "my_var" {
+            property.usage = PropertyUsageFlags::NO_EDITOR;
+            property.property_name = StringName::from("SuperNewTestPropertyName");
+            property.hint_info.hint_string = GString::from("SomePropertyHint");
+
+            // Makes no sense but allows to check if given ClassName can be properly moved to GDExtensionPropertyInfo
+            property.class_name = <ValidatePropertyTest as godot::obj::GodotClass>::class_name();
+        }
+    }
+}
+
+#[itest]
+fn validate_property_test() {
+    let obj = ValidatePropertyTest::new_alloc();
+    let properties: Array<Dictionary> = obj.get_property_list();
+
+    for property in properties.iter_shared() {
+        if property
+            .get("name")
+            .map(|v| v.to_string() == "SuperNewTestPropertyName")
+            .unwrap_or(false)
```

Comment:
The `.map().unwrap_or(false)` pattern can often be replaced with `is_some_and()`.

Reply:
And maybe this could also use an early exit (`continue`) to un-indent the following code 1 level?